packages/openapi-tanstack-query/src/index.js

```diff
@@ -0,0 +1,34 @@
+import { useMutation, useQuery } from "@tanstack/react-query";
+
+/**
+ * Create an openapi-fetch client.
+ * @type {import("./index.js").default}
+ */
+export default function createClient(client) {
```

Comment:
Just a note: if you can get away with just `src/index.ts` and that builds to `dist/index.js` that’ll be much simpler.

With some of the advanced TS inferences in `openapi-fetch`, it was more expedient to manage types separately from the runtime. But those scenarios are usually rare, and it’s always best to start with `.ts` files originally until they no longer work (that’s how `openapi-fetch` started, and it outgrew it).

Ultimately your call, though!

Reply:
I'll try my best to just directly go for typescript but I know that tanstack query types are really a pain to play with

Reply:
Understandable. If it makes sense to manage types separately then it’s no problem. Mainly my suggestion is because writing types separately from runtime increases the barrier to contribution for the average dev—it requires a more in-depth knowledge of TypeScript (which clearly you have, so this suggestion is more for contributor-friendliness than yourself 🙂). 

And right—if TanStack’s types lead to more indirection in `.ts` code then keeping them separate is 100% fine (`.ts` is an amazing tool to help people write safer code in general, but at higher levels if it leads to indirection in runtime code then it’s no longer improving quality and alternative methods should be sought).

Reply:
I've done some testing and I can definitely go full typescript.
Right now I have to do some `as any`  but it is related to the `openapi-fetch` package. I think by tweaking the types I will be able to have it fully properly typed.

Are you fine if I change a bit the openapi-fetch types?

Reply:
Sure, it should not do any breaking change